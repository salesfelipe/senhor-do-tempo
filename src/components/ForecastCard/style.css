.labelFont {
    font-size: 33% !important;
}

.ui.statistic {
    max-width: none;
}