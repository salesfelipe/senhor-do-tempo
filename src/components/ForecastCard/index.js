import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Segment, Card, Image, Label, Statistic, Grid } from 'semantic-ui-react';
import { GOOGLE_MAPS_KEY } from '../../utils/keys';
import './style.css';

const ForecastStats = ({ content, color, name, icon }) => (
    <Statistic>
        <Statistic.Value>
            <Label color={color} image className="labelFont">
                <Image src={icon} />
                {content}
            </Label>
        </Statistic.Value>
        <Statistic.Label>
            {name}
        </Statistic.Label>
    </Statistic>
)


class ForecastCard extends Component {
    render() {
        const { forecast, loading } = this.props;

        return (
            <Segment loading={loading}>
                <Card fluid>
                    {!loading &&
                        <iframe
                            src={`https://www.google.com/maps/embed/v1/view?key=${GOOGLE_MAPS_KEY}&zoom=10&center=${forecast.coord.lat},${forecast.coord.lon}`}
                            width="inherit"
                            height="inherit"
                            frameBorder="0"
                            title="city-map"
                            allowFullScreen />
                    }
                    <Card.Content>
                        <Segment>
                            <Grid columns="4">
                                <Grid.Column>
                                    <ForecastStats
                                        content={forecast.weather.description}
                                        color='grey'
                                        icon={forecast.weather.icon}
                                    />
                                </Grid.Column>
                                <Grid.Column>
                                    <ForecastStats
                                        content={forecast.stats.humidity}
                                        color='blue'
                                        name='Umidade'
                                    />
                                </Grid.Column>
                                <Grid.Column>
                                    <ForecastStats
                                        content={forecast.stats.tempMin}
                                        color='orange'
                                        name='Temp. Mínima'
                                    />
                                </Grid.Column>
                                <Grid.Column>
                                    <ForecastStats
                                        content={forecast.stats.tempMax}
                                        color='red'
                                        name='Temp. Máxima'
                                    />
                                </Grid.Column>
                            </Grid>
                        </Segment>
                    </Card.Content>
                </Card>
            </Segment>
        );
    }
}
ForecastCard.propTypes = {
    forecast: PropTypes.shape({
        coord: PropTypes.shape({
            lon: PropTypes.number,
            lat: PropTypes.number
        }),
        weather: PropTypes.shape({
            icon: PropTypes.string,
            description: PropTypes.string
        }),
        stats: PropTypes.shape({
            tempMax: PropTypes.string,
            tempMin: PropTypes.string,
            humidity: PropTypes.string
        })
    }),
    loading: PropTypes.bool
}

export default ForecastCard;