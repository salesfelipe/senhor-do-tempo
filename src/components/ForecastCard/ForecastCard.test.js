import React from 'react';
import ReactDOM from 'react-dom';
import ForecastCard from './index';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ForecastCard forecast={{coord:{}, weather:{}, stats: {}}}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly: loading false', () => {
  const tree = renderer
    .create(<ForecastCard forecast={{coord:{}, weather:{}, stats: {}, loading: false}}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly: loading true', () => {
  const tree = renderer
    .create(<ForecastCard forecast={{coord:{}, weather:{}, stats: {}, loading: true}}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});