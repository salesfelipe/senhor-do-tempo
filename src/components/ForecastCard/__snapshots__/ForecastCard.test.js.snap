// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui segment"
>
  <div
    className="ui fluid card"
    href={undefined}
    onClick={[Function]}
  >
    <iframe
      allowFullScreen={true}
      frameBorder="0"
      height="inherit"
      src="https://www.google.com/maps/embed/v1/view?key=AIzaSyCJ0RZIuECTJqLMj5fpzSxFORLHQahDI5I&zoom=10&center=undefined,undefined"
      title="city-map"
      width="inherit"
    />
    <div
      className="content"
    >
      <div
        className="ui segment"
      >
        <div
          className="ui grid"
        >
          <div
            className="seven wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui grey image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              />
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui blue image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Umidade
              </div>
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui orange image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Temp. Mínima
              </div>
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui red image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Temp. Máxima
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly: loading false 1`] = `
<div
  className="ui segment"
>
  <div
    className="ui fluid card"
    href={undefined}
    onClick={[Function]}
  >
    <iframe
      allowFullScreen={true}
      frameBorder="0"
      height="inherit"
      src="https://www.google.com/maps/embed/v1/view?key=AIzaSyCJ0RZIuECTJqLMj5fpzSxFORLHQahDI5I&zoom=10&center=undefined,undefined"
      title="city-map"
      width="inherit"
    />
    <div
      className="content"
    >
      <div
        className="ui segment"
      >
        <div
          className="ui grid"
        >
          <div
            className="seven wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui grey image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              />
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui blue image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Umidade
              </div>
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui orange image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Temp. Mínima
              </div>
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui red image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Temp. Máxima
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly: loading true 1`] = `
<div
  className="ui segment"
>
  <div
    className="ui fluid card"
    href={undefined}
    onClick={[Function]}
  >
    <iframe
      allowFullScreen={true}
      frameBorder="0"
      height="inherit"
      src="https://www.google.com/maps/embed/v1/view?key=AIzaSyCJ0RZIuECTJqLMj5fpzSxFORLHQahDI5I&zoom=10&center=undefined,undefined"
      title="city-map"
      width="inherit"
    />
    <div
      className="content"
    >
      <div
        className="ui segment"
      >
        <div
          className="ui grid"
        >
          <div
            className="seven wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui grey image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              />
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui blue image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Umidade
              </div>
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui orange image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Temp. Mínima
              </div>
            </div>
          </div>
          <div
            className="three wide column"
          >
            <div
              className="ui statistic"
            >
              <div
                className="value"
              >
                <div
                  className="ui red image label labelFont"
                  onClick={[Function]}
                >
                  <img
                    className="ui image"
                    src={undefined}
                  />
                </div>
              </div>
              <div
                className="label"
              >
                Temp. Máxima
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
